presto-main/src/main/java/com/facebook/presto/execution/warnings/DefaultWarningCollector.java

```diff
@@ -33,18 +38,34 @@
     @GuardedBy("this")
     private final Map<WarningCode, PrestoWarning> warnings = new LinkedHashMap<>();
     private final WarningCollectorConfig config;
+    private final Session session;
 
-    public DefaultWarningCollector(WarningCollectorConfig config)
+    public DefaultWarningCollector(WarningCollectorConfig config, Session session)
     {
         this.config = requireNonNull(config, "config is null");
+        this.session = session;
```

Comment:
I'll probably only keep the `WarningHandlingLevel` as member variable rather than the whole session.

Reply:
Ok will change this.